File: Java/Algorithm/InsertionSort.java
import java.util.Scanner;

public class InsertionSort {

	public static void main(String[] args) {
		
		Scanner s = new Scanner(System.in);
		System.out.print("Enter the No of Elements:");
		int n = s.nextInt();
		int[] intarray = new int[n];
		System.out.print("Enter the elements:" );
		for(int i=0;i<n;i++) {
			intarray[i] = s.nextInt();
		}
//Insertion sort Algorighm
		for (int unsortedindex=1;unsortedindex<intarray.length;unsortedindex++) {
			int newelement = intarray[unsortedindex];
			
			int i;
			for(i=unsortedindex; i>0 && intarray[i-1]>newelement;i--) {
				intarray[i] = intarray[i-1];
			}
			intarray[i] = newelement;
// You can remove these comments and watch the iterations for better understanding of this Algorithm			
//			for(int j =0;j<intarray.length;j++) {
//				System.out.print(intarray[j]+ "  ");
//			}
//			System.out.println();
		}
		System.out.println("\n");
		for(int i=0;i<intarray.length;i++) {
			System.out.print(intarray[i]+ " ");
		}
		
		s.close();
	}
	public static void swap(int[] array,int i,int j) {
		if(i==j) return;
		int temp=array[i];
		array[i]=array[j];
		array[j]=temp;
	}

}

File: C/Algorithms/Sorting/InsertionSort/InsertionSort.c
/* C Program for Insertion Sort */
#include <stdio.h>
int main()
{
    int a[100], number, i, j, temp;
    
    printf("\n Please Enter the total Number of Elements  :  ");
    scanf("%d", &number);
    
    printf("\n Please Enter the Array Elements  :  ");
    for(i = 0; i < number; i++)
        scanf("%d", &a[i]);
    
    for(i = 1; i <= number - 1; i++)
    {
        for(j = i; j > 0 && a[j - 1] > a[j]; j--)
        {
                temp = a[j];
                a[j] = a[j - 1];
                a[j - 1] = temp;
        }
    }
    printf("\n Insertion Sort Result : ");
    for(i = 0; i < number; i++)
    {
        printf(" %d \t", a[i]);
    }
    printf("\n");
    return 0;
}